src/OpenTelemetry.Instrumentation.Runtime/RuntimeMetrics.cs

```diff
@@ -102,6 +104,17 @@ public void Dispose()
             this.meter?.Dispose();
         }
 
+        private static IEnumerable<Measurement<long>> GetGarbageCollectionCounts()
+        {
+            Measurement<long>[] measurements = new Measurement<long>[NumberOfGenerations];
```

Comment:
This would cause allocation on each `Collect` cycle. We could use a dedicated static array in the class to store these measurements and reuse in each `Collect` cycle.

Reply:
I've moved it to static field. It's only allocated once during initialization.

Reply:
This metrics is an `Observable` and gets called once every 15~20 seconds. So it's not worth doing too much optimization for now. Most importantly, if there are multiple readers reading it at almost the same time, they could have incorrect values.